### PR TITLE
Remove ADSB mode from Plane

### PR DESCRIPTION
**Pull request by andyp1per**

### CODE REVIEW

**Review by magicrub — APPROVED** [review]
I'm happy with this as-is but it would be nice to at least add a comment to that switch. I'm also thinking that about half of those ``#if`` will disappear where its also checking mode_guided because that cane become a mode property like control_mode.is_guided() or control_mode.is_externally_controlled()

**Comment by magicrub** [comment]
hahaha I just now saw your second commit after writing my comment about to do just that! Nice work.

**Comment by tridge** [comment]
looks good. Have you done any testing of ADSB avoidance in SITL with these changes? We just need to make sure we don't break it

**Comment by andyp1per** [comment]
@tridge I've tried the SITL test - obviously all the avoidance parameters are gone so there is not a lot to test. It at least doesn't blow up when it receives the mavlink messages. Not sure what else I should be trying.

File: ArduPlane/control_modes.cpp [hunk 1]
```diff
@@ -41,8 +41,10 @@ Mode *Plane::mode_from_mode_num(const enum Mode::Number num)
         ret = &mode_loiter;
         break;
     case Mode::Number::AVOID_ADSB:
+#if HAL_ADSB_ENABLED
         ret = &mode_avoidADSB;
         break;
+#endif
```
[the remark this answers is not in the document]
Comment (andyp1per) [reply ArduPlane/control_modes.cpp:47]: Yeah I wasn't sure whether we ever even hit this, but selecting guided seemed a reasonable thing to do

**Comment by andyp1per** [comment]
Squashed

**Comment by tridge** [comment]
> I've tried the SITL test - obviously all the avoidance parameters are gone so there is not a lot to test.

why are they gone in SITL? We should still have ADSB in SITL

**Comment by andyp1per** [comment]
@tridge - sorry - I ran a specific test in SITL with HAL_ADSB_ENABLED as 0 to check things were still functioning ok. Normally in SITL the behavior is unchanged.

File: ArduPlane/mode.h [hunk 2]
```diff
@@ -352,10 +358,14 @@ class ModeAvoidADSB : public Mode
     void update() override;
 
     void navigate() override;
+
+    virtual bool is_guided_mode() const override { return true; }
```
Comment (magicrub) [inline ArduPlane/mode.h:362]: this is correct, at least for now, but another step in the onion will be to make this class inherit ModeGuided like it does in copter.

**Review by magicrub — COMMENTED** [review]
That switch statement needs a comment, otherwise LGTM!

File: ArduPlane/control_modes.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (andyp1per) [reply ArduPlane/control_modes.cpp:47]: Added a comment to show that it's intended behaviour

**Comment by andyp1per** [comment]
I've verified that switching to mode 14 with this compiled out simply puts the vehicle in guided mode

**Comment by Hwurzburg** [comment]
so which boards have ADSB and which do not now?

**Comment by andyp1per** [comment]
2MB boards only unless you explictly enable

**Comment by peterbarker** [comment]
On Thu, 24 Sep 2020, Andy Piper wrote:

> 2MB boards only unless you explictly enable

*Was* this available on 1MB boards?!

**Pull request merged**